Home work 5

TamirMandreev opened the pull request:
Выполнил 5-е домашнее задание

TamirMandreev committed:
Создал представление ProductCreateView для создания объекта модели Product

TamirMandreev committed:
Исправил ошибку Improperly Configured at /product_create/

В представление ProductCreateView добавил атрибут fields

TamirMandreev committed:
Создал шаблон отображения формы для модели Product

TamirMandreev committed:
Создал маршрут для ProductCreateView

TamirMandreev committed:
Создал автоматически генерируемую форму с использованием класса ModelForm и интегрировал ее в представление ProductCreateView

TamirMandreev committed:
Создал представление ProductUpdateView для редактирования объекта модели Product

TamirMandreev committed:
Добавил маршрут для представления ProductUpdateView

TamirMandreev committed:
В product_form.html добавил возможность загружать медиа файлы

enctype="multipart/form-data"

TamirMandreev committed:
Создал представление ProductDeleteView для удаления объекта модели Product

TamirMandreev committed:
Добавил маршрут для представления ProductDeleteView

TamirMandreev committed:
Создал шаблон для ProductDeleteView

TamirMandreev committed:
Добавил методы clean_name, clean_description и clean_price в ProductForm

painassasin commented on catalog/forms.py:
```diff
@@ -0,0 +1,59 @@
+    # Определить метод для валидации и очистки данных, введенных пользователем в поле description
+    def clean_description(self):
+        # Получить значение поля name
+        description = self.cleaned_data['description']
```
Описание может быть пустым (в базе null=True), а значит значение из словаря нужно получать через get и обрабатывать только если там не None
```python
description = self.cleaned_data.get('description')
if description:
    ...

TamirMandreev replied:
Готово. Теперь значение из словаря получаю через метод get и реализую проверку только если if description==True

TamirMandreev committed:
В ProductForm при валидации поля description (метод clean_description) могло быть вызвано исключение (ошибка), если поле пустое (None).

Чтобы этого недопустить, поле description теперь получаем через метод get и проверяем через условие if description
